commons-mongo/src/test/scala/com/avsystem/commons/mongo/BsonInputOutputTest.scala

```diff
@@ -0,0 +1,90 @@
+package com.avsystem.commons
+package mongo
+
+import java.nio.ByteBuffer
+
+import org.bson._
+import org.bson.io.BasicOutputBuffer
+import org.scalatest.FunSuite
+
+class BsonInputOutputTest extends FunSuite {
+  test("Roundtrip binary encoding of random objects") {
+    100 times {
```

Comment:
100 times random with simple assert - this just _asks_ to be a property based test (like [this one](https://github.com/AVSystem/scala-commons/blob/master/commons-redis/src/test/scala/com/avsystem/commons/redis/protocol/RedisMsgTest.scala#L22) for instance)

Reply:
Of course, my dude - this test wast just moved verbatim from another codebase, where I didn't have such goodies.